lib/ansible/modules/remote_management/ucs/ucs_uuid_pool.py

```diff
@@ -0,0 +1,198 @@
+
+    err = False
+
+    from ucsmsdk.mometa.uuidpool.UuidpoolPool import UuidpoolPool
+    from ucsmsdk.mometa.uuidpool.UuidpoolBlock import UuidpoolBlock
+
+    changed = False
+    try:
+        mo_exists = False
+        props_match = False
+        # dn is <org_dn>/uuid-pool-<name>
+        dn = module.params['org_dn'] + '/uuid-pool-' + module.params['name']
+        mo = ucs.login_handle.query_dn(dn)
+        if mo:
+            mo_exists = True
+
+        if module.params['state'] == 'absent':
+            if mo_exists:
+                if not module.check_mode:
+                    ucs.login_handle.remove_mo(mo)
+                    ucs.login_handle.commit()
+                changed = True
+        else:
+            if mo_exists:
+                # check top-level mo props
+                kwargs = dict(assignment_order=module.params['order'])
+                kwargs['descr'] = module.params['description']
+                if module.params['prefix']:
+                    kwargs['prefix'] = module.params['prefix']
+                if mo.check_prop_match(**kwargs):
+                    # top-level props match, check next level mo/props
+                    if module.params['last_uuid'] and module.params['first_uuid']:
+                        # uuid address block specified, check properties
+                        block_dn = dn + '/block-from-' + module.params['first_uuid'].upper() + '-to-' + module.params['last_uuid'].upper()
+                        mo_1 = ucs.login_handle.query_dn(block_dn)
+                        if mo_1:
+                            props_match = True
+                    else:
+                        # no UUID address block specified, but top-level props matched
+                        props_match = True
+
+            if not props_match:
+                if not module.check_mode:
+                    # create if mo does not already exist
+                    if not module.params['prefix']:
+                        module.params['prefix'] = 'derived'
+                    mo = UuidpoolPool(
+                        parent_mo_or_dn=module.params['org_dn'],
+                        name=module.params['name'],
+                        descr=module.params['description'],
+                        assignment_order=module.params['order'],
+                        prefix=module.params['prefix']
+                    )
+
+                    if module.params['last_uuid'] and module.params['first_uuid']:
+                        mo_1 = UuidpoolBlock(
+                            parent_mo_or_dn=mo,
+                            to=module.params['last_uuid'],
+                            r_from=module.params['first_uuid'],
+                        )
+
+                    ucs.login_handle.add_mo(mo, True)
+                    ucs.login_handle.commit()
+                changed = True
+
+    except Exception as e:
+        err = True
+        ucs.result['msg'] = "setup error: %s " % str(e)
+
+    ucs.result['changed'] = changed
```

Comment:
I wonder why you are doing this here, and not just use ucs.result['changed'] on assignment ?

Reply:
updated with ucs.result on assignment